Cucumber Branch containing features- Search, Sort and Pagination

Sharvaribarve opened the pull request:
This branch contains the automation tests for the features Search, Sort and Pagination

jeddcuison committed:
Initial Cucumber tests

Details:

- Use of DI via Spring
- Establish run step patterns
- Do cleanup after running each tests
- Ignore Test Suite (this will be deprecated soon)

jeddcuison committed:
Add README.md

Instructions on how to run tests

jeddcuison committed:
Delete group jvm table to prevent foreign key constraint error when deleting the group and the jvm data

jeddcuison committed:
Test media creation

jeddcuison committed:
Add file.upload.dir property

jeddcuison committed:
Rename method

jeddcuison committed:
Test web server creation

jeddcuison committed:
Test web app creation

Sharvaribarve committed:
Added tests and feature files for searching and sorting

Sharvaribarve committed:
Reverted change

Sharvaribarve committed:
Added media tests with attributes

Sharvaribarve committed:
Added pages for groups

Sharvaribarve committed:
Added test cases and formatted code

Sharvaribarve committed:
Merge branch 'master' into sharvariCucumberBranch

Sharvaribarve committed:
Removed unmerged comments from the file

Sharvaribarve committed:
Removed comment and deleted file

Sharvaribarve committed:
Changed formatting

Sharvaribarve committed:
Changed formatting

Sharvaribarve committed:
Changed formatting

Sharvaribarve committed:
Changed formatting

Sharvaribarve committed:
Update ManageMediaTest.java

Sharvaribarve committed:
Update ManageJvmTest.java

Sharvaribarve committed:
Update ManageWebAppTest.java

Sharvaribarve committed:
Update LoginRunSteps.java

Sharvaribarve committed:
Deleted Test.java file

Sharvaribarve committed:
Merge branch 'sharvariCucumberBranch' of https://github.com/cerner/jwala into sharvariCucumberBranch

Sharvaribarve committed:
Update ManageGroupTest.java

Sharvaribarve committed:
Deleted search file

Sharvaribarve committed:
Merge branch 'sharvariCucumberBranch' of https://github.com/cerner/jwala into sharvariCucumberBranch

Sharvaribarve committed:
Update manageWebApp.feature

Sharvaribarve committed:
Update manageWebApp.feature

Sharvaribarve committed:
Update manageWebApp.feature

Sharvaribarve committed:
Update ManageWebServerTest.java

Sharvaribarve committed:
Update ManageWebServerTest.java

Sharvaribarve committed:
Added formatting to JwalaUi

Sharvaribarve committed:
Merge branch 'sharvariCucumberBranch' of https://github.com/cerner/jwala into sharvariCucumberBranch

Sharvaribarve committed:
Removed new lines

Sharvaribarve committed:
deleted file and changed format

Sharvaribarve committed:
Removed new line

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/configuration/ManagePaginationTest.java:
```diff
@@ -0,0 +1,15 @@
+import org.junit.runner.RunWith;
+
+/**
+ * Created by SB053052 on 7/10/2017.
```
Change **all** SB053052  to your name since this is opensource

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManagePaginationRunSteps.java:
```diff
@@ -0,0 +1,59 @@
+public class ManagePaginationRunSteps {
+
+    @Autowired
+    JwalaUi jwalaUi;
```
Make this private

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManagePaginationRunSteps.java:
```diff
@@ -0,0 +1,59 @@
+    @When("I click the next page button")
+    public void clickRightButton() {
+        jwalaUi.clickWhenReady(By.className("ui-corner-right"));
+
```
Remove extra line

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManagePaginationRunSteps.java:
```diff
@@ -0,0 +1,59 @@
+    @Autowired
+    JwalaUi jwalaUi;
+
+    @When("I click the next page button")
```
Is this step being use ? If so Shouldn't this be

    @When("^I click the next page button$")

The rest of the steps in this class don't have ^ $ as well

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManageSearchRunSteps.java:
```diff
@@ -0,0 +1,33 @@
+    public void checkIfElementIsNotPresent(final String name) {
+        assertEquals(false, jwalaUi.isElementExists(By.xpath("//button[text()='" + name + "']")));
+    }
+
```
Remove extra spaces

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManageSearchRunSteps.java:
```diff
@@ -0,0 +1,33 @@
+        jwalaUi.clickWhenReady(By.xpath("//input"));
+        jwalaUi.sendKeys(searchString);
+    }
+
```
Remove extra space

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManageSortRunSteps.java:
```diff
@@ -0,0 +1,52 @@
+        jwalaUi.clickWhenReady(By.xpath("//span[contains(text(), '" + component + attribute + "')]"));
+    }
+
+
```
Remove extra space

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManageSortRunSteps.java:
```diff
@@ -0,0 +1,52 @@
+    @Autowired
+    JwalaUi jwalaUi;
+
+
```
Remove extra space

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManageSortRunSteps.java:
```diff
@@ -0,0 +1,52 @@
+    JwalaUi jwalaUi;
+
+
+    @When("I click on the sort button of component \"(.*)\" with attribute \"(.*)\"$")
```
You forgot to put ^

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManageSortRunSteps.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+
+    @When("I click on the sort button with attribute \"(.*)\"$")
```
You forgot to put ^

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManageSortRunSteps.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+
+    @Then("I see first item \"(.*)\"$")
```
You forgot to put ^

jeddcuison commented on jwala-integration-test/src/test/resources/com/cerner/jwala/ui/selenium/configuration/pagination.feature:
```diff
@@ -0,0 +1,280 @@
+      | statusPath         | /apache_pb.png      |
+    And I click on the sort button with attribute "Name"
+    And I see "29" in the webserver table
+
```
Where is When and Then ?

jeddcuison commented on jwala-integration-test/src/test/resources/com/cerner/jwala/ui/selenium/configuration/pagination.feature:
```diff
@@ -0,0 +1,280 @@
+    And I created a group with the name "A25"
+    And I created a group with the name "A26"
+    And I click the next page button
+    Then I see the text "Page 2/2"
```
No When ? Instead of having 2 "Then" we should just concatenate with And e.g. Then And And And ...

jeddcuison commented on jwala-integration-test/src/test/resources/com/cerner/jwala/ui/selenium/configuration/search.feature:
```diff
@@ -0,0 +1,130 @@
+    And I am in the group tab
+    And I created a group with the name "MMM"
+    And I created a group with the name "ZZZ"
+    And I fill in the search field with "MM"
```
Where's When ?

jeddcuison commented on jwala-integration-test/src/test/resources/com/cerner/jwala/ui/selenium/configuration/search.feature:
```diff
@@ -0,0 +1,130 @@
+      | name          | zzz    |
+      | webappContext | \name2 |
+      | group         | group2 |
+    And I fill in the search field with "app"
```
Where's When ?

jeddcuison commented on jwala-integration-test/src/test/resources/com/cerner/jwala/ui/selenium/configuration/search.feature:
```diff
@@ -0,0 +1,130 @@
+      | mediaType       | Apache HTTPD            |
+      | archiveFilename | apache-httpd-2.4.20.zip |
+      | remoteDir       | c:\ctp                  |
+    And I fill in the search field with "ap"
```
Where's When ?

jeddcuison commented on jwala-integration-test/src/test/resources/com/cerner/jwala/ui/selenium/configuration/search.feature:
```diff
@@ -0,0 +1,130 @@
+      | apacheHttpdMediaId | apache-httpd-2.4.20 |
+      | statusPath         | /apache_pb.png      |
+    And I am in the web server tab
+    And I fill in the search field with "my"
```
Where's When ?

jeddcuison commented on jwala-integration-test/src/test/resources/com/cerner/jwala/ui/selenium/configuration/search.feature:
```diff
@@ -0,0 +1,130 @@
+    And I don't see "aWebserver" in the table
+
+
+  Scenario: Search within  jvms
```
Is this complete ?

jeddcuison reviewed (changes requested):
Please see  my comments

Sharvaribarve committed:
Addressed code review comments by formatting and adhering to Cucumber standards

Sharvaribarve committed:
Removed extra lines

Sharvaribarve committed:
Update ManageSearchRunSteps.java

Sharvaribarve committed:
Removed extra lines

Sharvaribarve committed:
Added properties and made code attributes uniform

Sharvaribarve committed:
Merge branch 'sharvariCucumberBranch' of https://github.com/cerner/jwala into sharvariCucumberBranch

Sharvaribarve committed:
Modified code to load properties file only once

jmahmoodcc reviewed (commented):
Just a couple of general comments:
 - Please change the names of the JVMs, Web Servers, Web Apps, Groups, and Media to meaningful and unique names like jvm-1-test-name. In a lot of places there were just numbers or random letters used as names. When things fail this makes it a lot easier to trace and debug
 - Did we figure out a way to resolve things like server names and absolute paths yet? I see localhost and C:/ctp hard coded in a few places.

Sharvaribarve committed:
Addressed code review by standardizing names and changing paths

jeddcuison commented on jwala-integration-test/src/test/resources/com/cerner/jwala/ui/selenium/configuration/pagination.feature:
```diff
@@ -6,273 +6,274 @@ Feature: Pagination
     And I select the dropdown of "webserver" with option "100"
     And I created a web server with the following parameters:
-      | webserverName      | 1                   |
+      | webserverName      | 1WebServer          |
```
WebServer1, WebServer2, ... it more conventional. Judging from the name, attaching the WebServer part from the number required less effort hence the unconventional names. In addition the groups were renamed to Group1, Group2 etc, we should always be consistent on our naming convention.

jeddcuison reviewed (commented):
@jmahmoodcc, with regards to:

> Did we figure out a way to resolve things like server names and absolute paths yet? I see localhost and C:/ctp hard coded in a few places.

@Sharvaribarve changed the code and now gets some of the values from a properties file

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/SeleniumTestHelper.java:
```diff
@@ -22,6 +22,8 @@
 
     private static final String SELENIUM_PROPERTY_PATH = "selenium.property.path";
     private static final String TEST_PROPERTIES = "selenium/test.properties";
+    private static final String JWALA_PROPERTIES = "selenium/jwala.properties";
```
Please commit a sample property file just like test.properties so that it is easier for developers to digest the purpose and contents of the file

jeddcuison replied:
Can we make the property file name more descriptive like "parameters.properties" ?

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/SeleniumTestHelper.java:
```diff
@@ -22,6 +22,8 @@
     private static final String SELENIUM_PROPERTY_PATH = "selenium.property.path";
     private static final String TEST_PROPERTIES = "selenium/test.properties";
+    private static final String JWALA_PROPERTIES = "selenium/jwala.properties";
+    private static Properties jwalaProperties;
```
If you rename the properties file please don't forget to rename this one too

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/CommonRunSteps.java:
```diff
@@ -48,8 +74,60 @@ public void createMedia(final Map<String, String> parameters) {
+        createJvmRunSteps.setGroups(groups);
+        createJvmRunSteps.clickOkBtn();
+        createJvmRunSteps.waitForJvm(parameters.get("jvmName"));
+
```
Please remove extra line

jeddcuison commented on jwala-integration-test/src/test/java/com/cerner/jwala/ui/selenium/steps/configuration/ManagePaginationRunSteps.java:
```diff
@@ -0,0 +1,55 @@
+        assertNotNull(element);
+    }
+
+    @When("^I select the dropdown of \"(.*)\" with option \"(.*)\"$")
```
Please put this before "Then" so that you code structure follows given-when-then

jeddcuison commented on jwala-integration-test/src/test/resources/com/cerner/jwala/ui/selenium/configuration/search.feature:
```diff
@@ -0,0 +1,122 @@
+    And I am in the web apps tab
+    And I created a web app with the following parameters:
+      | webappName  | application |
+      | contextPath | \name1       |
```
use '/' and make the contextPath more descriptive e.g. "aContextPath1"  e.g.

    | contextPath | /name1 |

jeddcuison reviewed (commented):
Pass 1 complete please see comments, I'll do one more pass and build and run your tests before I tag my review as completed...

Sharvaribarve committed:
Addresed code review by changing naming and including sample properties file

jeddcuison committed:
Remove extra lines

jeddcuison committed:
Breakup long lines

jeddcuison committed:
Change ':' to '='

jeddcuison committed:
Simplify the sorting step and make it more readable

jeddcuison committed:
Remove remoteStpDir and rename remoteDir to media.remote.dir

jeddcuison committed:
Remove extra space

jeddcuison committed:
Remove unused step

jeddcuison committed:
Replace with non OS specific sample directory

jeddcuison committed:
Refactor the code

Details

- Extract common code into a method
- Revert JVM validation to just checking the table
- Replace wait for JVM method with check JVM

jeddcuison committed:
Use Spring to load the test step parameters property file

jeddcuison committed:
Change windows to win

jeddcuison committed:
Specify a longer wait time of 300 seconds instead of relying on the generic wait time

jeddcuison committed:
Format the code

jeddcuison committed:
Rename the classes

jeddcuison commented:
Already merged via pull request #150
